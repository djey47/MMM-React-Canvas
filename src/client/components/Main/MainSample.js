/* @flow */

import React, { PureComponent } from 'react';
import classnames from 'classnames';

import '../../styles/reset.scss';
import '../../styles/module.scss';
import './MainSample.scss';

type PropTypes = {};

type StateType = {};

/**
 * MagicMirror
 * Main REACT component sample
 * Rename this component and associated resources to your convenience
 */
class MainSample extends PureComponent<PropTypes, StateType> {
  static defaultProps = {}

  constructor(props: PropTypes) {
    super(props);
  }

  /**
   * @return component markup
   */
  render() {
    return (
      <div className={classnames('MainSample', 'dimmed', 'light', 'small')}>
        <p className="MainSample__description">
          Main component sample for MM2's React template
        </p>
      </div>
    );
  }
}

export default MainSample;
