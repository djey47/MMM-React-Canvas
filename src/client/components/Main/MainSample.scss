/** @define MainSample */

/**
 * MagicMirror
 * Style rules for Main component
 */

.MainSample {
  font-weight: bold;
}

.MainSample__description {
  font-style: italic;
}
