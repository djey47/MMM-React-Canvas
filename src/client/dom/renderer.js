/* @flow */

import React from 'react';
import ReactDOM from 'react-dom';

import MainSample from '../components/Main/MainSample';

/**
 * MagicMirror
 * HTML renderer
 */

/**
 * @returns HTML for main wrapper
 */
export const renderWrapper = (wrapperId: string): any => {
  const wrapper = document.createElement('div');
  wrapper.id = wrapperId;
  wrapper.className = wrapperId;

  return wrapper;
};

/**
 * REACT gateway helper
 * @return Mounted component
 */
export const renderMainComponent = (wrapperId: string): any => {
  const rootId = document.getElementById(wrapperId);
  if (!rootId) {
    Log.error(`** Could not find root div with id: ${wrapperId}! Aborting.`);
    return;
  }

  return ReactDOM.render(
    <MainSample />,
    rootId
  );
};
