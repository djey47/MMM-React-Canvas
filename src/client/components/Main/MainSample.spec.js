import React from 'react';
import renderer from 'react-test-renderer';
import MainSample from './MainSample';

describe('Main sample component', () => {
  it('should render correctly at initial state', () => {
    // given-when
    const component = renderer.create(
      <MainSample />,
    );

    // then
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
