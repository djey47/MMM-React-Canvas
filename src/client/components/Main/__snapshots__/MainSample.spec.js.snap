// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main sample component should render correctly at initial state 1`] = `
<div
  className="MainSample dimmed light small"
>
  <p
    className="MainSample__description"
  >
    Main component sample for MM2's React template
  </p>
</div>
`;
